addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1905,6 +1905,63 @@ options.registry.collapse = options.Class.extend({
     },
 });
 
+options.registry.MasonryLayout = options.Class.extend({
+    xmlDependencies: (options.Class.xmlDependencies || []).concat(
+        ['/website/static/src/snippets/s_masonry_block/s_masonry_block.xml']
+    ),
+
+    /**
+     * @override
+     */
+    start: function () {
+        if (!this.$target[0].dataset.currentTemplate) {
+            this.$target[0].dataset.currentTemplate = "default";
+            const $template = $(qweb.render(`website.s_template_masonry.default`, {}));
+            this.$('.container, .container-fluid, .o_container_small').append($template);
```

Comment:
If you do it that way, current users with the old masonry snippet structure will lose their entire snippet content just by going back in edit mode after the next version's release and clicking on their snippet :scream:

Probably using the `onBuilt` method to do that is exactly what you want.